Update Syslog docs (#78)
axiomhq/docs: send-data/syslog-proxy.mdx

sidebarTitle: Syslog
tags:
  ['axiom documentation', 'documentation', 'axiom', 'guide', 'syslog', 'syslog logs', 'tcp', 'udp', 'docker']
---

import Prerequisites from "/snippets/standard-prerequisites.mdx"
import Replacement from "/snippets/standard-replacement.mdx"

The Axiom Syslog Proxy acts as a syslog server to send data to Axiom.

<Note>
    The Axiom Syslog Proxy is an open-source project and welcomes your contributions. For more information, see the [GitHub repository](https://github.com/axiomhq/axiom-syslog-proxy).
</Note>

## Syslog limitations and recommended alternatives

Syslog is an outdated protocol from the 1980s. Some of the limitations are the following:

- Lack of error reporting and feedback mechanisms when issues occur.
- Inability to gracefully terminate the connection. This can result in missing data.

<Note>
For a more reliable and modern logging experience, consider using tools like [Vector](https://vector.dev/) to receive syslog messages and [forward them to Axiom](/send-data/vector). This approach bypasses many of syslog’s limitations.
</Note>

<Prerequisites />

Other requirements:

> - **Message size limit:** Axiom currently enforces an 8KB per-message size limit. This is in line with RFC5425 guidelines. Any message exceeding the limit causes the connection to close because Axiom doesn’t support ingesting truncated messages.

Reviewer: ```suggestion
- **Message size limit:** Axiom currently enforces a 64KB per-message size limit. This is in line with RFC5425 guidelines. Any message exceeding the limit causes the connection to close because Axiom doesn’t support ingesting truncated messages.
```

This was increased by @gordallott recently.

Author: Cool, thanks. I thought this was still in planning phase.

- **TLS requirement:** Axiom only supports syslog over TLS, specifically following RFC5425. Configure your syslog client accordingly.
- **Port requirements:** UDP log messages are sent on UDP port `514` to the Syslog server. TCP log messages are sent on TCP port `601` to the Syslog server.

> Ensure your messages conform to the size limit and TLS requirements. Frequent connection re-establishment may indicate that messages are rejected due to size or other formatting issues.

Author: ```suggestion
Ensure your messages conform to the size limit and TLS requirements. If the connection is frequently re-established and messages are rejected, the issue can be the size of the messages or other formatting issues.
```
